app/controllers/assignments_controller.rb

```diff
@@ -613,7 +613,10 @@ def set_repo_vars(assignment, grouping)
     grouping.group.access_repo do |repo|
       @revision = repo.get_revision_by_timestamp(Time.current, assignment.repository_folder)
       @last_modified_date = @revision&.server_timestamp
-      files = @revision.files_at_path(assignment.repository_folder)
+      files = @revision.tree_at_path(assignment.repository_folder)
+                       .select do |_, obj|
+                         obj.is_a?(Repository::RevisionFile) && !GitRepository.internal_file_names.include?(obj.name)
```

Comment:
It seems like you should only check for this when MarkUs is actually using git.

Reply:
Good catch, it should be Repository.get_class.internal_file_names to keep it repo-agnostic.

Reply:
Thanks for catching that. I will update this but please don't pull this in @david-yz-liu until PR #3790 is finished so I can rebase on top of that one.

Reply:
@mishaschwartz ready for you to rebase.